scala_proto/BUILD

```diff
@@ -0,0 +1,39 @@
+load("//scala_proto:scala_proto_toolchain.bzl", "scala_proto_toolchain")
+
+toolchain_type(
+    name = "toolchain_type",
```

Comment:
I freely admit to not understanding this stuff. It seems really weird to me the whole toolchain_type, toolchain, impl of the toolchain.... but I guess maybe they are mapping a typed system onto dynamic values in skylark.

Reply:
I don't really know about how much of this is required/needed tbh. I cargo culted this from the scala tool chain